src/sentry/static/sentry/app/views/organizationIntegrations/integrationListDirectory.tsx

```diff
@@ -271,41 +273,83 @@ export class IntegrationListDirectory extends AsyncComponent<
+
+    return {searchInput, selectedCategory};
+  };
+
+  /**
+   * Update the query string with the current filter parameter values.
+   */
+  updateQueryString = () => {
+    const {searchInput, selectedCategory} = this.state;
+
+    const searchString = queryString.stringify({
+      ...queryString.parse(this.props.location.search),
+      search: searchInput ? searchInput : undefined,
+      category: selectedCategory ? selectedCategory : undefined,
+    });
+
+    browserHistory.replace({
+      pathname: this.props.location.pathname,
+      search: searchString ? `?${searchString}` : undefined,
+    });
+  };
+
+  /**
+   * Filter the integrations list by ANDing together the search query and the category select.
+   */
+  getDisplayedList = (): AppOrProviderOrPlugin[] => {
+    const {fuzzy, list, searchInput, selectedCategory} = this.state;
+
+    let displayedList = list;
+
+    if (searchInput && fuzzy) {
+      const searchResults = fuzzy.search(searchInput);
+      displayedList = this.sortIntegrations(searchResults.map(i => i.item));
+    }
+
+    if (selectedCategory) {
+      displayedList = displayedList.filter(integration => {
+        return getCategoriesForIntegration(integration).includes(selectedCategory);
+      });
+    }
+
+    return displayedList;
+  };
+
   handleSearchChange = async (value: string) => {
     this.setState({searchInput: value}, () => {
-      if (!value) {
-        return this.setState({displayedList: this.state.list});
-      }
-      const result = this.state.fuzzy && this.state.fuzzy.search(value);
+      this.updateQueryString();
+      // TODO We're only re-calculating this list to track its size.
+      const result = this.getDisplayedList();
       this.debouncedTrackIntegrationSearch(value, result.length);
-      return this.setState({
-        displayedList: this.sortIntegrations(result.map(i => i.item)),
-      });
     });
   };
 
   onCategorySelect = ({value: category}: {value: string}) => {
     this.setState({selectedCategory: category}, () => {
-      if (!category) {
-        return this.setState({displayedList: this.state.list});
-      }
-      const result = this.state.list.filter(integration => {
-        return getCategoriesForIntegration(integration).includes(category);
-      });
-
-      return this.setState({displayedList: result}, () =>
-        trackIntegrationEvent(
-          {
-            eventKey: 'integrations.directory_category_selected',
-            eventName: 'Integrations: Directory Category Selected',
-            view: 'integrations_directory',
-            category,
-          },
-          this.props.organization
-        )
+      this.updateQueryString();
+
+      trackIntegrationEvent(
+        {
+          eventKey: 'integrations.directory_category_selected',
+          eventName: 'Integrations: Directory Category Selected',
+          view: 'integrations_directory',
+          category,
```

Comment:
@mgaeta We probably shouldn't be tracking a category select that's just an empty string

Reply:
Should I sub in "All Categories" or just not log?

Reply:
@mgaeta We weren't logging it before and I think that was fine